Orphan Maven/Go containers at the end of build (even successful) grab resources

I have experiences on a GKE deployed K8S + jx deployment that lead to have multiple maven & go containers continuing to grab memory & cpu, blocking new deployments at a given time.
Version :

> NAME               VERSION
> jx                 1.3.461
> jenkins x platform 0.0.2785
> Kubernetes cluster v1.9.7-gke.6
> kubectl            v1.11.3
> helm client        v2.9.1+g20adb27
> helm server        v2.10.0+g9ad53aa
> git                git version 2.13.5

In the maven logs I got : 
`INFO: Locating server among [http://jenkins:8080]
Oct 23, 2018 2:01:06 PM org.jenkinsci.remoting.engine.JnlpAgentEndpointResolver resolve
INFO: Remoting server accepts the following protocols: [JNLP4-connect, Ping]
Oct 23, 2018 2:01:06 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Agent discovery successful
  Agent address: jenkins-agent
  Agent port:    50000
  Identity:      2b:d8:02:3d:fc:ff:d6:d8:92:67:e6:5b:f8:34:54:aa
Oct 23, 2018 2:01:06 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Handshaking
Oct 23, 2018 2:01:06 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Connecting to jenkins-agent:50000
Oct 23, 2018 2:01:06 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Trying protocol: JNLP4-connect
Oct 23, 2018 2:01:06 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Remote identity confirmed: 2b:d8:02:3d:fc:ff:d6:d8:92:67:e6:5b:f8:34:54:aa
Oct 23, 2018 2:01:06 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Connected
Oct 23, 2018 2:01:46 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Terminated
Oct 23, 2018 2:01:46 PM hudson.remoting.UserRequest perform
WARNING: LinkageError while performing UserRequest:hudson.Launcher$RemoteLauncher$KillTask@3aa154ef
java.lang.NoClassDefFoundError: hudson/slaves/SlaveComputer
        at hudson.util.ProcessTree.get(ProcessTree.java:401)
        at hudson.Launcher$RemoteLauncher$KillTask.call(Launcher.java:1096)
        at hudson.Launcher$RemoteLauncher$KillTask.call(Launcher.java:1087)
        at hudson.remoting.UserRequest.perform(UserRequest.java:207)
        at hudson.remoting.UserRequest.perform(UserRequest.java:53)
        at hudson.remoting.Request$2.run(Request.java:358)
        at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at hudson.remoting.Engine$1$1.run(Engine.java:98)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: hudson.slaves.SlaveComputer
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at hudson.remoting.RemoteClassLoader.findClass(RemoteClassLoader.java:159)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 12 more

Oct 23, 2018 2:01:46 PM hudson.remoting.Request$2 run
WARNING: Failed to send back a reply to the request hudson.remoting.Request$2@56f56318
hudson.remoting.ChannelClosedException: channel is already closed
        at hudson.remoting.Channel.send(Channel.java:671)
        at hudson.remoting.Request$2.run(Request.java:370)
        at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at hudson.remoting.Engine$1$1.run(Engine.java:98)
        at java.lang.Thread.run(Thread.java:748)
Caused by: hudson.remoting.Channel$OrderlyShutdown
        at hudson.remoting.Channel$CloseCommand.execute(Channel.java:1225)
        at hudson.remoting.Channel$1.handle(Channel.java:560)
        at hudson.remoting.AbstractByteBufferCommandTransport.processCommand(AbstractByteBufferCommandTransport.java:202)
        at hudson.remoting.AbstractByteBufferCommandTransport.receive(AbstractByteBufferCommandTransport.java:188)
        at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer.onRead(ChannelApplicationLayer.java:186)
        at org.jenkinsci.remoting.protocol.ApplicationLayer.onRecv(ApplicationLayer.java:207)
        at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecv(ProtocolStack.java:669)
        at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processRead(SSLEngineFilterLayer.java:369)
        at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.onRecv(SSLEngineFilterLayer.java:117)
        at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecv(ProtocolStack.java:669)
        at org.jenkinsci.remoting.protocol.NetworkLayer.onRead(NetworkLayer.java:136)
        at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer.access$2200(BIONetworkLayer.java:48)
        at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Reader.run(BIONetworkLayer.java:283)
        ... 4 more
Caused by: Command close created at
        at hudson.remoting.Command.<init>(Command.java:68)
        at hudson.remoting.Channel$CloseCommand.<init>(Channel.java:1265)
        at hudson.remoting.Channel$CloseCommand.<init>(Channel.java:1263)
        at hudson.remoting.Channel.close(Channel.java:1436)
        at hudson.remoting.Channel.close(Channel.java:1403)
        at hudson.slaves.SlaveComputer.closeChannel(SlaveComputer.java:821)
        at hudson.slaves.SlaveComputer.kill(SlaveComputer.java:788)
        at hudson.model.AbstractCIBase.killComputer(AbstractCIBase.java:88)
        at hudson.model.AbstractCIBase.updateComputerList(AbstractCIBase.java:227)
        at jenkins.model.Jenkins.updateComputerList(Jenkins.java:1549)
        at jenkins.model.Nodes$6.run(Nodes.java:261)
        at hudson.model.Queue._withLock(Queue.java:1380)
        at hudson.model.Queue.withLock(Queue.java:1257)
        at jenkins.model.Nodes.removeNode(Nodes.java:252)
        at jenkins.model.Jenkins.removeNode(Jenkins.java:2063)
        at hudson.slaves.AbstractCloudSlave.terminate(AbstractCloudSlave.java:70)
        at org.jenkinsci.plugins.durabletask.executors.OnceRetentionStrategy$1$1.run(OnceRetentionStrategy.java:128)
        at hudson.model.Queue._withLock(Queue.java:1380)
        at hudson.model.Queue.withLock(Queue.java:1257)
        at org.jenkinsci.plugins.durabletask.executors.OnceRetentionStrategy$1.run(OnceRetentionStrategy.java:123)
        at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
        at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        ... 1 more

Oct 23, 2018 2:01:56 PM hudson.remoting.jnlp.Main$CuiListener error
SEVERE: jenkins/slaves/restarter/JnlpSlaveRestarterInstaller
java.lang.NoClassDefFoundError: jenkins/slaves/restarter/JnlpSlaveRestarterInstaller
        at jenkins.slaves.restarter.JnlpSlaveRestarterInstaller$FindEffectiveRestarters$1.onReconnect(JnlpSlaveRestarterInstaller.java:97)
        at hudson.remoting.EngineListenerSplitter.onReconnect(EngineListenerSplitter.java:49)
        at hudson.remoting.Engine.innerRun(Engine.java:647)
        at hudson.remoting.Engine.run(Engine.java:455)
Caused by: java.lang.ClassNotFoundException: jenkins.slaves.restarter.JnlpSlaveRestarterInstaller
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at hudson.remoting.RemoteClassLoader.findClass(RemoteClassLoader.java:159)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 4 more`

--- a/values.yaml
+++ b/values.yaml
@@ -545,7 +545,7 @@ jenkins:
     # Image: "jenkinsci/jenkins"
     # ImageTag: "2.89"
     Image: "jenkinsxio/jenkinsx"
-    ImageTag: "0.0.34"
+    ImageTag: "0.0.35"
     ImagePullPolicy: "IfNotPresent"
     Component: "jenkins-master"
     UseSecurity: true

--- a/values.yaml
+++ b/values.yaml
@@ -698,7 +698,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -740,7 +740,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -779,7 +779,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -821,7 +821,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -856,7 +856,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -890,7 +890,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -925,7 +925,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -960,7 +960,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -996,7 +996,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1032,7 +1032,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1066,7 +1066,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1100,7 +1100,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1136,7 +1136,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1172,7 +1172,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1208,7 +1208,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1244,7 +1244,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1279,7 +1279,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.14-1
+            Image: jenkinsci/jnlp-slave:3.26-1
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'

--- a/values.yaml
+++ b/values.yaml
@@ -698,7 +698,7 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -740,7 +740,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -779,7 +780,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -821,7 +823,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -856,7 +859,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -890,7 +894,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -925,7 +930,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -960,7 +966,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -996,7 +1003,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1032,7 +1040,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1066,7 +1075,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1100,7 +1110,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1136,7 +1147,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1172,7 +1184,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1208,7 +1221,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1244,7 +1258,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'
@@ -1279,7 +1294,8 @@ jenkins:
         ServiceAccount: jenkins
         Containers:
           Jnlp:
-            Image: jenkinsci/jnlp-slave:3.26-1
+            Image: jenkinsci/jnlp-slave:3.26-1-alpine
+
             RequestCpu: "100m"
             RequestMemory: "128Mi"
             Args: '${computer.jnlpmac} ${computer.name}'

